hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -432,45 +433,55 @@
     {
       name: chip_rv_dm_cpu_debug_mem
       desc: '''Verify access to the debug mem from the CPU.
+
+            - CPU programs RW type CSRs (such as progbuf0 and progbuf1) with random value.
+            - Read back to check their values for correctness.
```

Comment:
If the goal here is to access the debug memory (https://github.com/pulp-platform/riscv-dbg/blob/master/doc/debug-system.md#debug-memory), the core needs to be in debug mode first. Since we'll already access the debug memory as part of the more complete debug test (named chip_rv_dm_cpu_debug_req currently) below, you can either remove this test completely, or make it a negative test ("ensure that debug memory is not accessible while not in debug mode").

Reply:
Thanks Philipp.
I think it makes sense to make it a negative test. How can we know it's not accessible from a C program? Does it just return 0 when we read the debug memory?